iothub_client/src/iothub_message.c

```diff
@@ -348,6 +373,12 @@ IOTHUB_MESSAGE_HANDLE IoTHubMessage_Clone(IOTHUB_MESSAGE_HANDLE iotHubMessageHan
                 DestroyMessageData(result);
                 result = NULL;
             }
+            else if (source->creationTimeUtc != NULL && mallocAndStrcpy_s(&result->creationTimeUtc, source->creationTimeUtc) != 0)
+            {
+                LogError("unable to copy creationTimeUtc");
+                DestroyMessageData(result);
```

Comment:
DestroyMessageData should be changed to free ->creationTimeUtc, otherwise we will introduce a memory leak.
Also there should be a test failing somewhere without that change.